readme.md

```diff
@@ -47,13 +49,27 @@ const findUp = require('find-up');
 
 	console.log(await findUp(['rainbow.png', 'unicorn.png']));
 	//=> '/Users/sindresorhus/unicorn.png'
+
+	console.log(await findUp(dir => {
+		return fs.existsSync(path.join(dir, 'unicorn.png')) && 'foo';
```

Comment:
IIRC, I believe the reason I mixed sync and async here is because:

1. As you mentioned, async `fs.exists()` is deprecated.
2. At the time, converting from callbacks to promises would have been some extra noisy code to understand for any async `fs` method.

But now `fs.promises` is stable and we can use it to make a pretty simple example:

```js
const pathExists = filepath => fs.promises.access(filepath).then(_ => true).catch(_ => false);
console.log(await findUp(async (directory) => {
	const isInstalled = await pathExists(path.join(directory, 'node_modules'));
	return isInstalled && 'package.json';
}});
```

What do you think of that?

As for the utility methods, I agree they are useful and I'd also prefer for them to be properties on the `findUp` function. Let's do that. I think we should merge this PR first and then build the utilities on top of this functionality in a separate PR(s), since this is still useful with or without them.

Reply:
> How about something like this?

👍 

> I think we should merge this PR first and then build the utilities on top of this functionality in a separate PR(s), since this is still useful with or without them.

👍

Reply:
I realized that if we implement #33, we don't need the file/directory checks, so we can simply implement two utility methods => https://github.com/sindresorhus/find-up/issues/37